src/Microsoft.Sbom.Api/Providers/FilesProviders/FileListBasedFileToJsonProvider.cs

```diff
@@ -16,8 +19,13 @@ namespace Microsoft.Sbom.Api.Providers.FilesProviders
     /// </summary>
     public class FileListBasedFileToJsonProvider : PathBasedFileToJsonProviderBase
     {
-        [Inject]
-        public FileListEnumerator ListWalker { get; set; }
+        public FileListEnumerator ListWalker { get; }
+
+        public FileListBasedFileToJsonProvider(IConfiguration configuration, ChannelUtils channelUtils, ILogger log, FileHasher fileHasher, ManifestFolderFilterer fileFilterer, FileInfoWriter fileHashWriter, InternalSBOMFileInfoDeduplicator internalSBOMFileInfoDeduplicator, FileListEnumerator listWalker)
```

Comment:
NIT: Can you make all the constructors with long param lists to wrap around the next line?